File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max=Integer.MIN_VALUE;
        int curr_max=0;
        int i=0;
        int n = height.length;
        int j= n-1;
        while(i<j)
        {
            curr_max = (j-i)*Math.min(height[i],height[j]);
            if(curr_max > max)
            max = curr_max;
            if(height[i]<height[j])
            i++;
            else
            j--;
        }
        return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        total_max = 0
        currmax = 0
        left = 0
        right = len(height) -1
        while(left <= right):
            currmax = min(height[left], height[right]) * (right - left)
            total_max = max(total_max, currmax)
            if(height[left] < height[right]):
                left +=1
            elif(height[left] > height[right]):
                right -= 1
            else:
                left += 1
                right -=1
        return total_max